### PR TITLE
E2 e improve config watcher

### PR DESCRIPTION
**Pull request by GunaKKIBM**

### CODE REVIEW

**Comment by contsto** [comment]
:thumbsup: Coverage increased from [%] to [80.45%]

**Comment by contsto** [comment]
:thumbsup: Coverage increased from [%] to [80.8625%]

**Comment by contsto** [comment]
:thumbsup: Coverage increased from [%] to [81.875%]

File: pkg/ibmcsidriver/config_watcher.go [hunk 1]
```diff
@@ -72,22 +71,17 @@ func (cw *ConfigWatcher) updateSubnetList(oldObj, newObj interface{}) {
 		// Env variable VPC_SUBNET_IDS will be updated only when there is
 		// non empty data and there is change in configmap
 		if newConfig != "" && (newConfig != oldConfig) {
-			VPC_SUBNET_IDS := newData.Data[ConfigmapDataKey]
-			err := os.Setenv("VPC_SUBNET_IDS", VPC_SUBNET_IDS)
+			err := os.Setenv("VPC_SUBNET_IDS", newConfig)
```
Comment (skorati) [inline pkg/ibmcsidriver/config_watcher.go:74]: Change the variable names `newConfig` `oldConfig` to something that is easy to understand VPC_SUBNET_IDS_NEW/OLD OR similar.

File: pkg/ibmcsidriver/config_watcher.go [hunk 2]
```diff
@@ -72,22 +71,17 @@ func (cw *ConfigWatcher) updateSubnetList(oldObj, newObj interface{}) {
...
 			if err != nil {
-				cw.logger.Warn("Config watcher is unable to update the subnet list..", zap.Any("Updated list", VPC_SUBNET_IDS), zap.Error(err))
+				cw.logger.Warn("Config watcher is unable to update the subnet list..", zap.Any("Updated list", newConfig), zap.Error(err))
```
Comment (skorati) [inline pkg/ibmcsidriver/config_watcher.go:76]: This logging is wrong, easily misunderstood.
Could you please correct it, it was a miss from my side.

File: pkg/ibmcsidriver/config_watcher.go [hunk 3]
```diff
@@ -72,22 +71,17 @@ func (cw *ConfigWatcher) updateSubnetList(oldObj, newObj interface{}) {
...
 }
 
-func WatchClusterConfigMap(client k8sUtils.KubernetesClient, log *zap.Logger) {
-	configWatcher := NewConfigWatcher(client.Clientset, log)
-	if configWatcher == nil {
-		log.Error("ConfigWatcher not started ")
-		return
-	}
+func WatchClusterConfigMap(client rest.Interface, log *zap.Logger) {
+	configWatcher := NewConfigWatcher(client, log)
```
Comment (skorati) [inline pkg/ibmcsidriver/config_watcher.go:85]: please correct the logs in line 56, as we. discussed.

**Review by skorati — COMMENTED** [review]
Added some suggestions.

**Review by skorati — APPROVED** [review]
looks good.

**Review by sameshai — APPROVED** [review]

**Comment by contsto** [comment]
:thumbsup: Coverage increased from [%] to [81.625%]

**Pull request merged**